docs/configuredgraphfactory.txt

```diff
@@ -0,0 +1,504 @@
+Map<String, Object> map = new HashMap<String, Object>();
+map.put("storage.backend", "cassandrathrift");
+map.put("storage.hostname", "127.0.0.1");
+map.put("graph.graphname", "graph1");
+ConfigurationGraphManagement.getInstance().createConfiguration(new MapConfiguration(map));
+----
+
+Then you could access this graph on any JanusGraph node using:
+
+[source, gremlin]
+----
+ConfiguredGraphFactory.open("graph1");
+----
+
+[[template-configuration]]
+Template Configuration
+^^^^^^^^^^^^^^^^^^^^^^
+
+The `ConfigurationGraphManagement` also allows you to create one
+template configuration, which you can use to create many graphs using
+the same configuration template. For example:
+
+[source, gremlin]
+----
+Map<String, Object> map = new HashMap<String, Object>();
+map.put("storage.backend", "cassandrathrift");
+map.put("storage.hostname", "127.0.0.1");
+ConfigurationGraphManagement.getInstance().createTemplateConfiguration(new MapConfiguration(map));
+----
+
+After doing this, you can create graphs using the template
+configuration:
+
+[source, gremlin]
+----
+ConfiguredGraphFactory.create("graph2");
+----
+
+This method will first create a new configuration for "graph2" by
+copying over all the properties associated with the template
+configuration and storing it on a configuration for this specific graph.
+This means that this graph can be accessed in, on any JanusGraph node,
+in the future by doing:
+
+[source, gremlin]
+----
+ConfiguredGraphFactory.open("graph2");
+----
+
+[[updating-configurations]]
+Updating Configurations
+^^^^^^^^^^^^^^^^^^^^^^^
+
+[[important]]
+Important
+^^^^^^^^^
+
+All interactions with both the `JanusGraphFactory` and the
+`ConfiguredGraphFactory` that interact with configurations that define
+the property "graph.graphname" go through the `JanusGraphManager` which
+keeps track of graph references created on the given JVM. Think of it as
+a graph cache. For this reason:
+
+IMPORTANT: Any updates to a configuration are not guaranteed to take effect until
+you remove the graph in question on every janusgraph node in your
+cluster.
+
+You can do so by calling:
+
+[source, gremlin]
+----
+ConfiguredGraphFactory.close("graph2");
+----
+
+Since graphs created using the template configuration first create a
+configuration for that graph in question using a copy and create method,
+this means that:
+
+IMPORTANT: Any updates to a specific graph created using the template
+configuration are not guaranteed to take effect on the specific graph
+until_:
+
+[source, gremlin]
+----
+1. The relevant configuration is removed: `ConfigurationGraphManagement.getInstance().removeConfiguration("graph2");
+2. The graph in question has been closed on every JanusGraph node: `ConfiguredGraphFactory.close("graph2");
+3. The graph is recreated using the template configuration: `ConfiguredGraphFactory.create("graph2");
+----
+
+[[update-examples]]
+Update Examples
+^^^^^^^^^^^^^^^
+
+1.  We migrated our Cassandra data to a new server with a new
```

Comment:
It looks like the Asciidoc is getting confused by the code blocks in between numbered bullets. Try using `1)` instead of `1.` or just leave out the numbering.